src/storage_common.c

```diff
@@ -43,10 +43,11 @@ static void thingset_storage_save_handler(struct k_work *work)
     else {
         LOG_WRN("Data not stored because previous load failed.");
     }
-
+#ifdef CONFIG_THINGSET_STORAGE_AUTOSAVE
     if (IS_ENABLED(CONFIG_THINGSET_STORAGE_AUTOSAVE)) {
         thingset_sdk_reschedule_work(dwork, K_HOURS(CONFIG_THINGSET_STORAGE_AUTOSAVE_INTERVAL));
     }
+#endif
```

Comment:
The problem is not the `CONFIG_THINGSET_STORAGE_AUTOSAVE`, but the `_INTERVAL` which depends on `CONFIG_THINGSET_STORAGE_AUTOSAVE`, right?

Compile-time check makes sense in that case. But let's remove the runtime check, as we don't need both.

```suggestion

#ifdef CONFIG_THINGSET_STORAGE_AUTOSAVE
    thingset_sdk_reschedule_work(dwork, K_HOURS(CONFIG_THINGSET_STORAGE_AUTOSAVE_INTERVAL));
#endif
```